src/extra/AutoUi/Collection/index.tsx

```diff
@@ -0,0 +1,385 @@
+import { NoRecordsFoundArrow } from './NoRecordsFoundArrow';
+
+// Assumptions that I think we can easily make:
+// We only handle a single-level schema. If a schema is nested, it is handled by the `format` component internally.
+
+// TODO:
+// In rendition if there is an existing notification with a certain ID when submitting a new one, replace it with the new one.
+// Move tableWithCustomColumns to rendition
+// Make the Table JSON-schema compliant
+// Implement sort functions per format
+// 1-to-1 relationships should resolve as an object, not an array with one item (eg is_for__device_type should be an object when expanded).
+// One to many relationships can be both a number (when using $count) and an array of objects, pine should return counts as part of an object instead (eg. always return an object with __count, and data: []). This can generalize for any field, if we find a `__count` we can render the count instead. Maybe handle this as a format, so it will handle both numbers and arrays? I guess this will be the smallest shape of a collection summary.
+// Make table handle tags better so we don't have to filter it out from the schema.
+// Specifically for the table, how should we treat columns that are based on multiple schema fields, shall we have some rule around that? Eg. the release "deployment" column combines "status" and "source" and it does appear more compact than having a separate column just to say it is a cloud build or a balena deploy one.
+// How shall we handle batch edits (especially when the data differs between entries)? We already do this for HUP but it is all custom code, we need to generalize it.
+// SDK's CRUD should all have the same function signature (eg. apiKey.create doesnt take a single object, but two separate fields).
+
+const HeaderGrid = styled(Flex)`
+	margin-left: -4px;
+	margin-right: -4px;
+
+	> * {
+		margin-left: 4px;
+		margin-right: 4px;
+	}
+`;
+
+const getSelectedItems = <T extends BaseResource<T>>(
+	newItems: T[],
+	selectedItems: T[],
+) => {
+	if (!size(selectedItems)) {
+		return selectedItems;
+	}
+	// update the selections
+	selectedItems = intersectionBy(newItems, selectedItems, 'id');
+	return selectedItems;
+};
+
+export interface ActionData<T> {
+	action: AutoUIAction<T>;
+	schema: JSONSchema;
+	affectedEntries?: T[];
+}
+
+export interface CollectionProps<T> {
+	/** Model is the property that describe the data to display with a JSON structure */
+	model: AutoUIModel<T>;
+	/** Array of data to display */
+	data: T[] | undefined;
+	/** Formats are custom widgets to render in the table cell. The type of format to display is described in the model. */
+	formats?: Format[];
+	/** Actions is an array of actions applicable on the selected items */
+	actions?: Array<AutoUIAction<T>>;
+	/** The sdk is used to pass the method to handle tags when added removed or updated */
+	sdk?: {
+		tags?: ResourceTagModelService;
+		refresh?: () => void;
```

Comment:
How about having `refresh` as a separate top level property?
That will allow us to drop an extra useMemo from the UI and semantically I'm not sure it should be part of the sdk.